### PR TITLE
Created transaction documentation

### PR DESCRIPTION
**Pull request by Craz1k0ek**
Added documentation for the transaction syntax, based on the 3.0 documentation and current implementation.
Added the Star model as example model, as it is used throughout the Fluent examples.

### CODE REVIEW

**Comment by Craz1k0ek** [comment]
Possibly fixes [this issue](https://github.com/vapor/docs/issues/531)

File: 4.0/docs/fluent/transaction.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Transactions
+
+Transactions allow you to ensure multiple operations complete succesfully before saving data to your database. 
+Once a transaction is started, you may run Fluent queries normally. However, no data will be saved to the database until the transaction completes. 
+If an error is thrown at any point during the transaction (by you or the database), none of the changes will take effect.
+
+To perform a transcation, you need access to something that can connect to the database.
+This is usually an incoming HTTP request. Use the `req.db.transaction(_ closure:)` method.
```
Comment (0xTim) [inline 4.0/docs/fluent/transaction.md:8]: Nit formatting and tighten up the language
```suggestion
To perform a transcation, you need access to something that can connect to the database. This is usually an incoming HTTP request. For this, use `req.db.transaction(_ :)`:
```

File: 4.0/docs/fluent/transaction.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```swift
+req.db.transaction { (database) -> EventLoopFuture<Void> in
+    // use database
+}
+```
+Once inside the transaction closure, you must use the supplied database (named `database` in the example) to perform queries.
+
+The closure expects an `EventLoopFuture<T>` return value. Once this future completes succesfully, the transaction will be commited.
```
Comment (0xTim) [inline 4.0/docs/fluent/transaction.md:16]: Nit remove
```suggestion
Once this closure returns successfully, the transaction will be committed.
```

File: 4.0/docs/fluent/transaction.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+return req.db.transaction { (database) -> EventLoopFuture<Void> in
+    return sun.save(on: database).flatMap { _ in
+        return sirius.save(on: database)
+    }
+}
+```
+The above example will save `sun` and *then* `sirius` before completing the transaction. If either star fails to save, neither will save.
+
+Once the transaction is completed, the result can be transformed to, for example, a simple HTTP status response to indicate completion as shown below.
```
Comment (0xTim) [inline 4.0/docs/fluent/transaction.md:29]: Nit remove passive voice
```suggestion
Once the transaction completes, the result can be transformed into a different future, for example into a HTTP status to indicate completion as shown below:
```

File: 4.0/docs/fluent/transaction.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+```swift
+return req.db.transaction { (database) -> EventLoopFuture<Void> in
```
Comment (0xTim) [inline 4.0/docs/fluent/transaction.md:31]: Nit - match the style guide

```suggestion
return req.db.transaction { database in
```

File: 4.0/docs/fluent/transaction.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+}
+```
+Once inside the transaction closure, you must use the supplied database (named `database` in the example) to perform queries.
+
+The closure expects an `EventLoopFuture<T>` return value. Once this future completes succesfully, the transaction will be commited.
+```swift
+var sun: Star = ...
+var sirius: Star = ...
+
+return req.db.transaction { (database) -> EventLoopFuture<Void> in
```
Comment (0xTim) [inline 4.0/docs/fluent/transaction.md:21]: Nit match the style guide
```suggestion
return req.db.transaction { database in
```

File: 4.0/docs/fluent/transaction.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# Transactions
+
+Transactions allow you to ensure multiple operations complete succesfully before saving data to your database. 
+Once a transaction is started, you may run Fluent queries normally. However, no data will be saved to the database until the transaction completes. 
+If an error is thrown at any point during the transaction (by you or the database), none of the changes will take effect.
+
+To perform a transcation, you need access to something that can connect to the database.
+This is usually an incoming HTTP request. Use the `req.db.transaction(_ closure:)` method.
+```swift
+req.db.transaction { (database) -> EventLoopFuture<Void> in
```
Comment (0xTim) [inline 4.0/docs/fluent/transaction.md:10]: Nit match style guide

```suggestion
req.db.transaction { database in
```

File: 4.0/docs/fluent/transaction.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    // use database
+}
+```
+Once inside the transaction closure, you must use the supplied database (named `database` in the example) to perform queries.
```
Comment (0xTim) [inline 4.0/docs/fluent/transaction.md:14]: Clarify where the database comes from
```suggestion
Once inside the transaction closure, you must use the database supplied in the closure parameter (named `database` in the example) to perform queries.
```

**Review by 0xTim — CHANGES_REQUESTED** [review]
Great start! Just a few nits to resolve

**Review by 0xTim — APPROVED** [review]
Thanks!

**Pull request merged**